File: src/main/java/testCodi/a01binaryGap/Solution.cs
namespace testCodi.a01BinaryGap
{
    using System;
    // you can also use other imports, for example:
    // using System.Collections.Generic;

    // you can write to stdout for debugging purposes, e.g.
    // Console.WriteLine("this is a debug message");

    public class Solution
    {
        public int solution(int N)
        {
            // write your code in C# 6.0 with .NET 4.5 (Mono)
            string binary = Convert.ToString(N, 2);
            var maxGap = 0;
            var previousOne = binary.IndexOf('1');
            if (-1 == previousOne)
            {
                return maxGap;
            }
            binary = binary.Substring(previousOne + 1);

            while (binary.Length > 0)
            {
                int currentOne = binary.IndexOf('1');
                if (-1 != currentOne)
                {
                    if (maxGap < currentOne)
                    {
                        maxGap = currentOne;
                    }
                    binary = binary.Substring(currentOne + 1);
                    previousOne = currentOne;
                }
                else
                {
                    break;
                }
            }
            return maxGap;
        }
    }

}

File: src/main/java/testCodi/a01binaryGap/solution.rb
# you can write to stdout for debugging purposes, e.g.
# puts "this is a debug message"

def solution_binary_gap(n)
    # write your code in Ruby 2.2
  binary = n.to_s(2)
  # puts "original: " + n.to_s
  # puts "original binary: " + binary
  max_gap = 0
  previous_one = binary.index('1')
  if -1 == previous_one
    return max_gap
  end
  binary = binary[previous_one + 1..-1]
  # puts "binary: " + binary

  while binary.length > 0
    # puts binary
    current_one = binary.index('1')
    if nil != current_one
      if max_gap < current_one
        max_gap = current_one
      end
      binary = binary[current_one + 1..-1]
      previous_one = current_one
    else
      break
    end
  end
  max_gap
end
